spec/features/projects/life_cycle/active_in_project_spec.rb

```diff
@@ -0,0 +1,135 @@
+  shared_let(:closing_stage) { create(:project_stage_definition, name: "Closing") }
+
+  let(:project_life_cycle_page) { Pages::Projects::Settings::LifeCycle.new(project) }
+
+  context "with sufficient permissions" do
+    current_user { user_with_permission }
+
+    it "allows toggling the active/inactive state of lifecycle steps and filtering them" do
+      project_life_cycle_page.visit!
+
+      project_life_cycle_page.expect_listed(initiating_stage => false,
+                                            ready_to_execute_gate => false,
+                                            executing_stage => false,
+                                            ready_to_close_gate => false,
+                                            closing_stage => false)
+
+      # Activate the stages to be found within the project
+      project_life_cycle_page.toggle(initiating_stage)
+      project_life_cycle_page.toggle(ready_to_close_gate)
+      project_life_cycle_page.toggle(closing_stage)
+
+      wait_for_network_idle
+
+      # Expect the activation state to be kept after a reload
+      visit home_path
+      project_life_cycle_page.visit!
+
+      project_life_cycle_page.expect_listed(initiating_stage => true,
+                                            ready_to_execute_gate => false,
+                                            executing_stage => false,
+                                            ready_to_close_gate => true,
+                                            closing_stage => true)
+
+      # Disable all stages at once
+      project_life_cycle_page.disable_all
+
+      wait_for_network_idle
+
+      # Expect the activation state to be kept after a reload
+      visit home_path
+      project_life_cycle_page.visit!
+
+      project_life_cycle_page.expect_listed(initiating_stage => false,
+                                            ready_to_execute_gate => false,
+                                            executing_stage => false,
+                                            ready_to_close_gate => false,
+                                            closing_stage => false)
+
+      # Enable all stages at once
+      project_life_cycle_page.enable_all
+
+      wait_for_network_idle
+
+      # Expect the activation state to be kept after a reload
+      visit home_path
+      project_life_cycle_page.visit!
+
+      project_life_cycle_page.expect_listed(initiating_stage => true,
+                                            ready_to_execute_gate => true,
+                                            executing_stage => true,
+                                            ready_to_close_gate => true,
+                                            closing_stage => true)
+
+      # The user can filter the life cycle steps
+      project_life_cycle_page.filter_by("Clo")
+
+      project_life_cycle_page.expect_listed(ready_to_close_gate => true,
+                                            closing_stage => true)
```

Comment:
This doesn't check that others are not visible

Reply:
Thanks for that pointer. It uncovered a bug in the filtering I relied on.